provisioning/service/src/Config/EnrollmentGroup.cs

```diff
@@ -258,14 +258,21 @@ public Attestation Attestation
             }
             set
             {
-                if(!((value ?? throw new ArgumentNullException(nameof(value))) is X509Attestation))
+                if (value == null)
                 {
-                    throw new ArgumentException("Attestation for enrollmentGroup shall be X509");
+                    throw new ArgumentNullException(nameof(value));
+                }
+                else if (!(value is X509Attestation) && !(value is SymmetricKeyAttestation))
+                {
+                    throw new ArgumentException("Attestation for enrollmentGroup shall be X509 or symmetric key");
                 }
 
-                if((((X509Attestation)value).RootCertificates == null) && (((X509Attestation)value).CAReferences == null))
+                if (value is X509Attestation)
```

Comment:
in latest c# there is an option to do the following and the assignment and variable creation happens automatically. Not sure if this will work in here:
`if (value as X509Attestation x509Attestation) { // use x509Attestation variable here }`

Reply:
or better yet, let service do this validation :)

---
In reply to: [248149239](https://github.com/Azure/azure-iot-sdk-csharp/pull/750#discussion_r248149239) [](ancestors = 248149239)